src/ussd_optout.js

```diff
@@ -1,25 +1,155 @@
+                    : $('Please let us know why you do not want MomConnect messages');
+
+                return new ChoiceState(name, {
+                    question:  question,
+
+                    choices: [
+                        new Choice("miscarriage", $("Miscarriage")),
+                        new Choice("stillbirth", $("Baby was stillborn")),
+                        new Choice("babyloss", $("Baby died")),
+                        new Choice("not_useful", $("Messages not useful")),
+                        new Choice("other", $("Other"))
+                    ],
+
+                    next: function(choice) {
+                        if (_.contains(["not_useful", "other"], choice.value)){
+                            return "states_end_no_enter";
+                        } else {
+                            return "states_subscribe_option";
+                        }
+                    }
+                });
+            });
+        });
+
+        self.states.add("states_subscribe_option", function(name) {
+            return new ChoiceState(name, {
+                question: $("We are sorry for your loss. Would you like " +
+                            "to receive a small set of free messages " +
+                            "to help you in this difficult time?"),
+
+                choices: [
+                    new Choice("states_end_yes", $("Yes")),
+                    new Choice("states_end_no_enter", $("No"))
+                ],
+
                 next: function(choice) {
-                    return "state_start";
+                    if (choice.value == "states_end_yes") {
+                        return hub
+                        .create_change(
+                            {
+                                "registrant_id": self.im.user.answers.operator.id,
+                                "action": "momconnect_loss_switch",
+                                "data": {
+                                    "reason": self.im.user.answers.state_start
+                                }
+                            }
+                        )
+                        .then(function() {
+                            return choice.value;
+                        });
+                    } else {
+                        return choice.value;
+                    }
                 }
             });
         });
 
+        self.states.add("states_end_no_enter", function(name) {
+            var optout_info = {
+                "optout_type": "stop",
+                "identity": self.im.user.answers.operator.id,
+                "reason": self.im.user.answers.state_start,
+                "address_type": "msisdn",
+                "address": self.im.user.answers.operator_msisdn,
+                "request_source": "ussd_optout",
+                "requestor_source_id": self.im.config.testing_message_id || self.im.msg.message_id
+            };
+            return is
+            .optout(optout_info)
```

Comment:
Please remove the `is.optout` code (and the corresponding fixture). Depending on the optout reason, this should be doing a `create_change` with `"action": "momconnect_loss_optout"` or `"action": "momconnect_nonloss_optout". It may be easiest to have to different interstitial states for those two cases - so instead of having `states_end_no_enter`, have `state_send_loss_optout` and `state_send_nonloss_optout`.